src/mlpack/core/tree/cosine_tree/cosine_tree.cpp

```diff
@@ -150,6 +150,196 @@ CosineTree::CosineTree(const arma::mat& dataset,
   ConstructBasis(treeQueue);
 }
 
+//! Copy the given tree.
+CosineTree::CosineTree(const CosineTree& other) :
+    dataset(other.parent->GetDataset()),
+    delta(other.delta),
+    parent(other.Parent()),
+    left(other.Left()),
+    right(other.Right()),
```

Comment:
Since you're modifying these later, you could just set them to `NULL` here.  I think that would be more clear.

Reply:
Do you mean `delta` as well or just `left`,`right` and `parent`? Because `delta` isn't being modified from what I can see.

Reply:
Oh, just `left`, `right`, and `parent`.  It looks good now, thanks. :+1: